packages/core/src/tracing/instrumentation/messaging/bull.js

```diff
@@ -227,9 +212,21 @@ function instrumentedProcessJob(ctx, originalProcessJob, originalArgs) {
     const spanT = attributes.X_INSTANA_T;
     const spanP = attributes.X_INSTANA_S;
 
+    const parentSpan = cls.getCurrentSpan();
+
+    if ((parentSpan && parentSpan.p === spanP && parentSpan.t === spanT) || (parentSpan && parentSpan.n !== 'bull')) {
+      // We allow a new entry span even if there is already an active entry span, because repeatable and bulked jobs can
+      // run in parallel if concurrency when enabled.
+      // But here we check if the job parent span data is the same as the existent span. In this case, we don't
```

Comment:
> run in parallel if concurrency when enabled.

Should it be "is enabled"?

> But here we check if the job parent span data is the same as the existent span

I wasn't able to understand the use case here. 

I had a look at bull's [codebase](https://github.com/OptimalBits/bull/blob/019d6125bea8a195d7690d52707b4e7089a92f16/lib/queue.js#L874) and it looks to me they execute `processJobs` n times (number of concurrencies) and if a job repeatable, it (including the concurrency config) is executed again at a time t.

Does that mean if concurrency is enabled, we don't want to create multiple entry spans for a specific job?

Thanks :)

Reply:
Ah, thanks. It was indeed `is`, not `when`.
This condition was transitioned from `blocking instrumentation if ANY type of span already exists` to `allow instrumenting Bull spans that are not exactly the same as the existing one, if any`.

The reason for the change is: when concurrency is enabled, repeatable and bulked jobs can be sent in parallel, depending on the number of child process are being used. When this happens, they can be received at the same time, which was causing some spans to be lost (as we were blocking ANY kind of existing span).

The condition here is more to have _some_ level of filtering in regarding of spans to be instrumented.

I added this explanation in the PR description. Maybe I wasn't clear enough.

Reply:
I missed the removal of https://github.com/instana/nodejs/pull/412/files#diff-9e34bfdddf61bfb8623b8f23fec0279bee4aae47c0d2d2845493ba61182bcff4L166

I just noticed!

Okay now it makes sense. And thanks for the details <3